aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -0,0 +1,144 @@
+// Basic functions on sparse tensors
+
+#include <ATen/ATen.h>
+#include <ATen/InitialTensorOptions.h>
+#include <ATen/Layout.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/Parallel.h>
+#include <ATen/SparseCsrTensorImpl.h>
+#include <ATen/SparseCsrTensorUtils.h>
+#include <ATen/SparseTensorImpl.h>
+
+namespace at {
+namespace native {
+
+using namespace at::sparse;
+
+// Construction of CSR tensors.
+SparseTensor new_csr_tensor(const TensorOptions& options) {
+  // TODO: remove this comment after enabling autograd support for CSR tensor
+  // constructor.
+  // TORCH_INTERNAL_ASSERT(impl::variable_excluded_from_dispatch());
+  AT_ASSERT(options.layout() == kSparseCsr);
+  DispatchKey dispatch_key;
+  if (options.device().is_cuda()) {
+    dispatch_key = DispatchKey::SparseCsrCUDA;
+  } else {
+    dispatch_key = DispatchKey::SparseCsrCPU;
+  }
+
+  return detail::make_tensor<SparseCsrTensorImpl>(
+      DispatchKeySet(dispatch_key), options.dtype());
```

Comment:
Everything else from options is being ignored; you should probably check that they are all nullopt